### PR TITLE
update algo.take to solve #59177

### LINKED ISSUE #59177: BUG: pd.api.extensions.take does not accept NumpyExtensionArray
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd 
import numpy as np

arr = pd.arrays.NumpyExtensionArray(np.array([1,2,3]))
pd.api.extensions.take(arr,[2])

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\mambaforge\envs\test_py300\Lib\site-packages\pandas\core\algorithms.py", line 1166, in take
    raise TypeError(
TypeError: pd.api.extensions.take requires a numpy.ndarray, ExtensionArray, Index, or Series, got NumpyExtensionArray.
```

### Issue Description

I'd expect NumpyExtensionArray to be considered an ExtensionArray, so compatable with pd.api.extensions.take.

### Expected Behavior

return
```
<NumpyExtensionArray>
[2]
Length: 1, dtype: int32
```

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by mutricyl**
- [x] closes #59177
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mutricyl** [comment]
need to work on 

FAILED pandas/tests/dtypes/test_generic.py::TestABCClasses::test_abc_hierarchy[ABCNumpyExtensionArray-inst6-ABCExtensionArray-subs2]

**Comment by mutricyl** [comment]
There are a few others `if isinstance(xxx, (ABCIndex, ABCSeries, ExtensionArray))` like tests in pandas code (algorithms.py, common.py, categorial.py, blocks.py) and I wonder if I should find them all and append `ABCNumpyExtensionArray` ?

File: pandas/core/dtypes/generic.py [hunk 1]
```diff
@@ -139,7 +139,13 @@ def _subclasscheck(cls, inst) -> bool:
         "ABCExtensionArray",
         "_typ",
         # Note: IntervalArray and SparseArray are included bc they have _typ="extension"
-        {"extension", "categorical", "periodarray", "datetimearray", "timedeltaarray"},
+        {
+            "extension",
```
Comment (mroeschke) [inline pandas/core/dtypes/generic.py:143]: Am I correct in seeing that just for formatting here changed? i.e. There is no new elements added to this set?

Comment (mutricyl) [reply pandas/core/dtypes/generic.py:143]: ## short story

Indeed the set content have not changed.

## long story

I first tried to resolve the issue adding one element to the set which induces black to change formatting. I realized afterwards that ABCNumpyExtensionArray already existed so I finaly used it. I forgot at the end to revert the formating change for ABCExtensionArray

Comment (mroeschke) [reply pandas/core/dtypes/generic.py:143]: Thanks, could you undo the formatting here?

**Review by mroeschke — APPROVED** [review]

**Pull request merged**